iOS/SceneDelegate.swift

```diff
@@ -89,6 +92,96 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
 		coordinator.cleanUp(conditional: conditional)
 	}
 	
+	// Handle Opening of URLs
+	
+	func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
+		DispatchQueue.main.async {
+			for context in URLContexts {
+				// Show Unread View or Article
+				if context.url.absoluteString.contains(WidgetDeepLink.unread.url.absoluteString) {
+					guard let comps = URLComponents(string: context.url.absoluteString ) else { return  }
+					let id = comps.queryItems?.first(where: { $0.name == "id" })?.value
+					if id != nil {
+						if AccountManager.shared.isSuspended {
+							AccountManager.shared.resumeAll()
+						}
+						let articles = try! AccountManager.shared.fetchArticles(.unread)
+						let article = articles.filter({ $0.articleID == id }).first
+						
```

Comment:
You don't need to fetch all the articles just to tell the coordinator which to select.  It can select based on articleID.  Let's expose the `selectArticleInCurrentFeed` function in `SceneCoordinator` and use that instead.  More on that coming...

Reply:
Fixed.